apps/test/integration/levelSolutions/applab/ec_thumbnail.js

```diff
@@ -104,17 +103,17 @@ export default {
         sinon.stub(console, 'log');
 
         tickWrapper.runOnAppTick(Applab, CAPTURE_TICK_COUNT + 1, () => {
-          expect(isCaptureComplete()).to.be.false;
+          assert.isFalse(isCaptureComplete());
           tickWrapper
             .tickAppUntil(Applab, isCaptureComplete)
             .then(() => {
-              expect(project.saveThumbnail).not.to.have.been.called;
+              assert(project.saveThumbnail.notCalled);
```

Comment:
my extremely vague memory is that one of these commands (`expect` or `assert`) gives completely useless error output while the other gives at least some hint as to where the problem might be. before merging this, would you be willing to try out having one failing `assert` and one failing `expect` inside a tickWrapper to make sure we're not reducing usefulness here?

Reply:
That's a good point. I can tell you from memory that we _are_ reducing usefulness here; `expect(x).not.to.have.been.called` is specific and will report a correspondingly specific failure, while `assert(x)` will give you an "unspecified assertion failure." (This is part of the value of the `sinon-chai` library.)

I'll address this.  I can:

- Add messages to these assertions so they fail usefully (maybe more usefully than before).
- Try to reintroduce `sinon-chai` for this specific test.
- Move to `reconfiguredChai` for this test.

I'm leaning toward the first option.  Any preferences?

Reply:
Can you not simply `import {expect} from 'chai'`?

Reply:
Oh, I assume that would require reintroducing `sinon-chai`.

What's our long-term vision for our test dependencies? Specifically, are we intending to continue to support _both_ `expect` and `assert` in our JS tests? If so, then the first option seems reasonable but I really don't love the idea of continuing to do that.

Reply:
I expect to continue to support both, in part because I find each more expressive in different situations - in particular due to the pain point that `assert(false)` throws but `expect(false)` does not.

But deciding about sinon-chai can be independent of the assert/expect decision; you can also `assert.calledOnce` with sinon-chai.  I'm asking whether it's better to have no extra assertion dependencies in these tests, or whether we care.

Reply:
I like the idea of replacing `configuredChai` (chai-enzyme) with `sinon-chai` here instead of moving to `assert` because it preserves the benefit of `expect` over `assert`, namely being able tell why your test failed from the test output.

As for whether to standardize on `expect` or continue to allow `expect` and `assert` to coexist, one idea would be to "soft standardize" on expect meaning you should prefer it unless you need some particular behavior associated only with `assert`.

Apologies if I've misunderstood a question or ignored a concern!

Reply:
Okay, I'm convinced that I should drop these changes and just use `reconfiguredChai` in this file until we have a more coherent strategy around our test assertions.  Commit incoming...